From f376b46203b88fab3227bd533f3c2b156c138f26 Mon Sep 17 00:00:00 2001
From: Mahesh Mahadevan <r9aadq@freescale.com>
Date: Thu, 13 Mar 2008 15:26:21 -0500
Subject: [PATCH] ENGR00068609 Update to use UART as wakeup source

Update the code to use UART as wakeup source, the changes are needed as
the serial core driver now includes the code for wakeup

Signed-off-by: Mahesh Mahadevan <r9aadq@freescale.com>
---
 arch/arm/mach-mx3/serial.h  |    5 +++--
 arch/arm/mach-mx37/serial.h |    5 +++--
 drivers/serial/mxc_uart.c   |   14 ++------------
 3 files changed, 8 insertions(+), 16 deletions(-)

diff --git a/arch/arm/mach-mx3/serial.h b/arch/arm/mach-mx3/serial.h
index 3374692..c93b168 100644
--- a/arch/arm/mach-mx3/serial.h
+++ b/arch/arm/mach-mx3/serial.h
@@ -1,5 +1,5 @@
 /*
- * Copyright 2004-2007 Freescale Semiconductor, Inc. All Rights Reserved.
+ * Copyright 2004-2008 Freescale Semiconductor, Inc. All Rights Reserved.
  */
 
 /*
@@ -27,7 +27,8 @@
  * hardware flow control (set this option to 0) or hardware-driven hardware
  * flow control (set this option to 1).
  */
-#define UART1_HW_FLOW           1
+/* UART used as wakeup source */
+#define UART1_HW_FLOW           0
 /*!
  * This specifies the threshold at which the CTS pin is deasserted by the
  * RXFIFO. Set this value in Decimal to anything from 0 to 32 for
diff --git a/arch/arm/mach-mx37/serial.h b/arch/arm/mach-mx37/serial.h
index 926d44d..2404715 100644
--- a/arch/arm/mach-mx37/serial.h
+++ b/arch/arm/mach-mx37/serial.h
@@ -1,5 +1,5 @@
 /*
- * Copyright 2004-2007 Freescale Semiconductor, Inc. All Rights Reserved.
+ * Copyright 2004-2008 Freescale Semiconductor, Inc. All Rights Reserved.
  */
 
 /*
@@ -22,7 +22,8 @@
  * hardware flow control (set this option to 0) or hardware-driven hardware
  * flow control (set this option to 1).
  */
-#define UART1_HW_FLOW           1
+/* UART used as wakeup source */
+#define UART1_HW_FLOW           0
 /*!
  * This specifies the threshold at which the CTS pin is deasserted by the
  * RXFIFO. Set this value in Decimal to anything from 0 to 32 for
diff --git a/drivers/serial/mxc_uart.c b/drivers/serial/mxc_uart.c
index 642a560..e03ee08 100644
--- a/drivers/serial/mxc_uart.c
+++ b/drivers/serial/mxc_uart.c
@@ -1,5 +1,5 @@
 /*
- * Copyright 2004-2007 Freescale Semiconductor, Inc. All Rights Reserved.
+ * Copyright 2004-2008 Freescale Semiconductor, Inc. All Rights Reserved.
  */
 
 /*
@@ -1791,12 +1791,6 @@ static int mxcuart_suspend(struct platform_device *pdev, pm_message_t state)
 		umxc->port.info->tty->hw_stopped = 1;
 	}
 
-	if (device_may_wakeup(&pdev->dev)) {
-		/* UART RTS signal is used as wakeup source */
-		writel(MXC_UARTUCR1_RTSDEN, umxc->port.membase + MXC_UARTUCR1);
-		gpio_uart_active(umxc->port.line, umxc->ir_mode);
-	}
-
 	return 0;
 }
 
@@ -1820,10 +1814,7 @@ static int mxcuart_resume(struct platform_device *pdev)
 	if (umxc->port.info && umxc->port.info->flags & UIF_INITIALIZED) {
 		umxc->port.info->tty->hw_stopped = 0;
 	}
-	if (device_may_wakeup(&pdev->dev)) {
-		writel(0, umxc->port.membase + MXC_UARTUCR1);
-		gpio_uart_inactive(umxc->port.line, umxc->ir_mode);
-	}
+
 	uart_resume_port(&mxc_reg, &umxc->port);
 
 	return 0;
@@ -1863,7 +1854,6 @@ static int mxcuart_probe(struct platform_device *pdev)
 
 		uart_add_one_port(&mxc_reg, &mxc_ports[id]->port);
 		platform_set_drvdata(pdev, mxc_ports[id]);
-		pdev->dev.power.can_wakeup = 1;
 	}
 	return 0;
 }
-- 
1.5.4.4

